packages/sdk/src/services/MetaMaskSDK/InitializerManager/setupDappMetadata.ts

```diff
@@ -15,7 +15,8 @@ export function setupDappMetadata(instance: MetaMaskSDK) {
   const { options } = instance;
 
   // Check if iconUrl and url are valid
-  const urlPattern = /^https?:\/\//u; // Regular expression for URLs starting with http:// or https://
+  // eslint-disable-next-line require-unicode-regexp
+  const urlPattern = /^(http|https):\/\/[^\s]*|localhost(:\d+)?$/; // Regular expression for URLs starting with http:// or https://
```

Comment:
why do we need localhost?

Reply:
also can we fix it without the unicode regexp issue?

Reply:
On this case:
url: window.location.host